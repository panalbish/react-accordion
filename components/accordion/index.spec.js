import test from 'ava';
import React from 'react';
import { shallow } from 'enzyme';
import render from 'react-test-renderer';

import Accordion from '../accordion';

const Content = () => (
  <div className="container1">
    <h1>This is a content title for accordion</h1>
    <p>
      Lorem ipsum dolor sit amet, consetetur sadipscing elitr,
      sed diam nonumy eirmod tempor invidunt ut labore et dolore
       magna aliquyam erat, sed diam voluptua.
    </p>
  </div>
);

const defaultComponent = () => (
  <Accordion title="Dummy Accordion">
    <Content />
  </Accordion>
);

test('renders the default Accordion Component', (t) => {
  const accordionTree = render.create(defaultComponent()).toJSON();
  t.snapshot(accordionTree);
  const wrapper = shallow(defaultComponent());
  t.is(wrapper.instance().props.title, 'Dummy Accordion');
  t.is(wrapper.instance().props.alternateTitle, '');
  t.is(wrapper.instance().props.initialPanelState, false);
  t.is(wrapper.instance().state.open, false);
});

test('show the accordion content with alternative title when clicked', (t) => {
  const wrapper = shallow(
    <Accordion title="Dummy Accordion" alternateTitle="I am different" >
      <Content />
    </Accordion>
  );
  t.is(wrapper.instance().state.open, false);
  t.is(wrapper.instance().props.alternateTitle, 'I am different');
  wrapper.find('.title').simulate('click');
  t.is(wrapper.instance().state.open, true);
  t.is(wrapper.find('.title-text').text(), 'I am different');
});
