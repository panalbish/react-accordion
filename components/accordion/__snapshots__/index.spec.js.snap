exports[`renders the default Accordion Component 1`] = `"{\"__ava_react_jsx\":{\"type\":\"div\",\"props\":{\"className\":\"accordion\"},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"title\"},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"arrow-wrapper\"},\"children\":[{\"type\":\"i\",\"props\":{\"className\":\"fa fa-angle-down\"},\"children\":null}]},{\"type\":\"span\",\"props\":{\"className\":\"title-text\"},\"children\":[\"Dummy Accordion\"]}]},{\"type\":\"div\",\"props\":{\"className\":\"content\"},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"content-text\"},\"children\":[{\"type\":\"div\",\"props\":{\"className\":\"container1\"},\"children\":[{\"type\":\"h1\",\"props\":{},\"children\":[\"This is a content title for accordion\"]},{\"type\":\"p\",\"props\":{},\"children\":[\"Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.\"]}]}]}]}]}}"`;
